app/health/select.go

```diff
@@ -0,0 +1,137 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package health
+
+import (
+	"regexp"
+
+	pb "github.com/prometheus/client_model/go"
+)
+
+type labelSelector func(*pb.MetricFamily) *pb.Metric
+
+func maxLabel(metricsFam *pb.MetricFamily) *pb.Metric {
+	var max *pb.Metric
+	for _, metric := range metricsFam.Metric {
+		if max == nil {
+			max = metric
+		}
+
+		var val float64
+		switch metricsFam.GetType() {
+		case pb.MetricType_COUNTER:
+			val = metric.Counter.GetValue()
+		case pb.MetricType_GAUGE:
+			val = metric.Gauge.GetValue()
+		default:
+			panic("invalid metric type for simple value labelSelector")
+		}
+
+		if val > max.Counter.GetValue() {
+			max = metric
+		}
+	}
+
+	return max
+}
+
+func countNonZeroLabels(metricsFam *pb.MetricFamily) *pb.Metric {
+	gauge := &pb.Metric{
+		Gauge:       new(pb.Gauge),
+		TimestampMs: metricsFam.Metric[0].TimestampMs,
+	}
+
+	for _, metric := range metricsFam.Metric {
+		if metric.Gauge.GetValue() != 0 {
+			*guage.Gauge.Value++
+		}
+	}
+
+	return gauge
+}
+
+func noLabels(metricsFam *pb.MetricFamily) *pb.Metric {
+	if len(metricsFam.Metric) != 1 {
+		panic("multiple metrics for no label labelSelector")
+	}
+
+	return metricsFam.Metric[0]
+}
+
+func sumLabels(metricsFam *pb.MetricFamily) *pb.Metric {
+	sum := new(pb.Metric)
+	for _, metric := range metricsFam.Metric {
+		switch metricsFam.GetType() {
+		case pb.MetricType_COUNTER:
+			*sum.Counter.Value += metric.Counter.GetValue()
+		case pb.MetricType_GAUGE:
+			*sum.Gauge.Value += metric.Gauge.GetValue()
+		case pb.MetricType_HISTOGRAM:
+			panic("histogram sum not implemented yet")
+		default:
+			panic("invalid metric type for simple value labelSelector")
+		}
+	}
+
+	return sum
+}
+
+func l(name, val string) *pb.LabelPair {
```

Comment:
can we have a proper name of this function?

Reply:
no 😄